app/cryptography/token_helper.py

```
@@ -0,0 +1,97 @@
+import json
+
+from jwcrypto import jwe, jwt
+from jwcrypto.common import base64url_decode
+from jwcrypto.jwe import InvalidJWEData
+from jwcrypto.jws import InvalidJWSSignature, InvalidJWSObject
+from jwcrypto.jwt import JWTInvalidClaimFormat, JWTMissingClaim, JWTExpired
+
+from app.authentication.invalid_token_exception import InvalidTokenException
+
+
+def decrypt_jwe(encrypted_token, secret_store, purpose, default_kid=None):
+    try:
+        jwe_token = jwe.JWE(algs=['RSA-OAEP', 'A256GCM'])
+        jwe_token.deserialize(encrypted_token)
+
+        jwe_kid = extract_kid_from_header(encrypted_token, default_kid)
+
+        sr_private_jwk = secret_store.get_private_key_by_kid(purpose, jwe_kid)
+
+        jwe_token.decrypt(sr_private_jwk)
+
+        return jwe_token.payload.decode()
+    except InvalidJWEData as e:
+        raise InvalidTokenException(repr(e))
```

you can do `raise InvalidTokenException from e` which chains the exceptions together

---

Thats a lot neater. The code i had was pulled from the old code.

---

However your code doesn't work